File: Swift/palindrome-number.swift
/**
 * Problem Link: https://leetcode.com/problems/palindrome-number/
 *
 *
 */

class Solution {
    func isPalindrome(_ x: Int) -> Bool {
        if x < 0 || (x != 0 && x % 10 == 0) {
            return false
        }
        var xx = x
        var yy = 0
        while (xx > yy) {
            let c = xx % 10
            yy = yy * 10 + c
            xx /= 10
        }
        return xx == yy || (xx == yy / 10)
    }
}

File: Python/palindrome-number.py
/**
 * https://leetcode.com/problems/palindrome-number/
 * 
 * 
 */ 
// Date: Thu Mar 27 17:04:19 PDT 2025
class Solution:
    def isPalindrome(self, x: int) -> bool:
        text = str(x)
        start = 0
        end = len(text) - 1
        while start < end:
            if text[start] != text[end]:
                return False
            start += 1
            end -= 1
        return True
